subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskExecuterResult.java

```diff
@@ -16,17 +16,41 @@
 
 package org.gradle.api.internal.tasks;
 
+import com.google.common.collect.ImmutableList;
 import org.gradle.caching.internal.origin.OriginMetadata;
+import org.gradle.internal.Try;
+import org.gradle.internal.execution.ExecutionOutcome;
 
+import java.util.List;
 import java.util.Optional;
 
 public interface TaskExecuterResult {
+    /**
+     * Returns the reasons for executing this task. An empty list means the task was not executed.
+     */
+    List<String> getExecutionReasons();
+
+    /**
+     * The outcome of executing the task.
+     */
+    Try<ExecutionOutcome> getOutcome();
+
     /**
      * If the execution resulted in some previous output being reused, this returns its origin metadata.
      */
     Optional<OriginMetadata> getReusedOutputOriginMetadata();
 
     TaskExecuterResult NO_REUSED_OUTPUT = new TaskExecuterResult() {
```

Comment:
I think the name of this constant is a little bit strange. `NO_REUSED_OUTPUT` sounds like the task did not re-use any output. For me that means that the task executed. It also says above "An empty list means the task was not executed". This is in contradiction with the outcome "ExecutionOutcome.EXECUTED_FULLY".

I don't really understand what the executer result means :(

Reply:
OK, I simplified this by only providing information whether or not the task was executed incrementally, as this is the only thing we need to understand at the call site.

Reply:
I still don't understand what `NO_REUSED_OUTPUT` means, I think we should use a better name here. This executer result is not that the task didn't re-use any output. Any task that executes doesn't re-use the output. This is about the task not even trying to re-use the outputs.

How about `SKIPPED_WITHOUT_OUTPUTS` or just `WITHOUT_OUTPUTS`?

Reply:
`WITHOUT_OUTPUTS` it is.